src/gfx/geometries/ExtrudedObjectsGeometry.js

```diff
@@ -85,13 +85,66 @@ class ExtrudedObjectsGeometry extends ChunkedObjectsGeometry {
         positions[vtxIdx + 1] = point.y;
         positions[vtxIdx + 2] = point.z;
 
-        tmpPrev.subVectors(point, prevPt).normalize();
-        tmpNext.subVectors(point, nextPt).normalize();
-        tmpPrev.add(tmpNext).normalize();
+        // Counting normals:
+        // - No slope
+        //   Radius doesn't change throught part => normals are parallel with the plane contains section points
+        //   normal = vToPrevPointInSection + vToNextPointInSection (all vectors are scaled for being 1 in length)
+        // - Slope
+        //   Radius changes throught part => normals aren't parallel with the plane contains section points
+        //   normal = vTangentInSectionPlane x vToSuchPointInPrevSection (all vectors are scaled for being 1 in length)
+        if (hasSlope) {
+          if (i !== 1) {
```

Comment:
Correct normals are great! Now, we should fix the code climate a bit. 
At first, I think `if (i==1)` is more readable. Please, swap the` if` branches.